pkg/updater/updater.go

```diff
@@ -244,16 +201,19 @@ func (u *updaterImpl) Start(_ context.Context) error {
 }
 
 // Stop stops the garbage collector.
-func (u *updaterImpl) Stop(_ context.Context) error {
+func (u *updaterImpl) Stop(ctx context.Context) error {
 	u.rc.Close()
+	u.telemetry.Stop(ctx)
 	close(u.stopChan)
 	u.requestsWG.Wait()
 	close(u.requests)
 	return nil
 }
 
 // Bootstrap installs the stable version of the package.
-func (u *updaterImpl) Bootstrap(ctx context.Context, pkg string) error {
+func (u *updaterImpl) BootstrapDefault(ctx context.Context, pkg string) (err error) {
+	span := tracer.StartSpan("BootstrapDefault")
+	defer func() { span.Finish(tracer.WithError(err)) }()
```

Comment:
Shouldn't you you use [StartSpanFromContext](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1@v1.62.0/ddtrace/tracer#StartSpanFromContext) to propagate the trace with the context?